src/realm/sync/subscriptions.cpp

```diff
@@ -798,6 +798,47 @@ std::vector<SubscriptionSet> SubscriptionStore::get_pending_subscriptions() cons
     return subscriptions_to_recover;
 }
 
+void SubscriptionStore::notify_all_state_change_notifications(Status error)
```

Comment:
nit: since this can be `Status::OK()` (as mentioned in a comment), maybe the parameter name could be simply status